### DISCUSSION TITLE
Multi-platform different base images

### DISCUSSION THREAD

**Discussion by masip85** [docker/buildx#1928] [Q&A] [answered]
Hello

I am carrying out different machine learning processes. Depending on arm64 or amd64 I use completely different base images, so, one of them uses CUDA and the other doesn't. All steps after the from are the same in different platforms

I've been reading carefully this : (https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/)
But, it works because when `FROM --platform=$BUILDPLATFORM alpine AS build` is executed, for both platforms, its image name is the same. What happens when for two different platforms base image names are completely different?
It's also a problem, when for different platforms wget I use depends on completely different paths. I can use a build-arg for the url/path, but, I cannot correlate a build-arg depending on the platform.

Yes, I can execute both with different scripts with different build-args, create a manifest,amend and so on.  But, is there a way to do this straight forward? I've been exploring bakes, but I see the same problem: I can't correlate platform to an argument.

Am I missing something?

**Comment by jedevc** [comment, marked as answer]
You can do this by interpolating `BUILDPLATFORM` (or similar args) into the stage name. For example:

```dockerfile
FROM --platform=linux/amd64 base1 as build-amd64
FROM --platform=linux/arm64 base2 as build-arm64

FROM build-$BUILDARCH
# the rest of your build
```

You can see an example of this kind of style here: https://github.com/tonistiigi/dockerfile-alpine/blob/main/Dockerfile

**Reply by masip85** [reply]
Does exist a possibility like this, when I have this issue:

For different architechtures url are very very different.
' ' ' 
FROM --platform=linux/amd64 base1 as build-amd64
FROM --platform=linux/arm64 base2 as build-arm64

FROM build-$BUILDARCH
[...]
RUN wget "AAAAAAAA_url_download_for_arm64" (for amd64 could be BBBBBBB_url_download.for_amd64)
RUN  ... build .. 
[...]
' ' ' 

could be an argument, but I can't make it depends on the platform

**Reply by crazy-max** [reply]
I guess you want smth like https://github.com/crazy-max/docker-docker/blob/8c24b23faa7847f3455fc5ff531404ddf95c262c/Dockerfile-23.0#L18-L27 ?
